File: frontend/src/app/service/curatorship/curatorship.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Curatorship } from '../../component/curatorship/curatorship';
import { Observable } from 'rxjs';

const BASE_URL = ["http://localhost:8080/api/manager"]

@Injectable({
  providedIn: 'root'
})
export class CuratorshipService {

  constructor(private http: HttpClient) { }
  
  fetchAllCuratorships():Observable<Curatorship[]> {
    return this.http.get<Curatorship[]>(BASE_URL + `/curatorship`);
  }
  
  createCuratorship(data: Curatorship) {
    return this.http.post<Curatorship>(BASE_URL + `/curatorship`, data);
  }
  
  updateCuratorship(data: Curatorship) {
    return this.http.put<Curatorship>(BASE_URL + `/curatorship/${data.id_curatorship}`, data);
  }
  
  deleteCuratorship(id_curatorship: Number) {
    return this.http.delete<Curatorship>(BASE_URL + `/curatorship/${id_curatorship}`);
  }

}

File: backend/src/main/java/com/g0urd/grad_project/service/CuratorshipService/CuratorshipService.java
package com.g0urd.grad_project.service.CuratorshipService;

import java.util.List;

import com.g0urd.grad_project.models.Curatorship;

public interface CuratorshipService {

    List<Curatorship> fetchAllCuratorships();

    Curatorship findById(Long id);

    Curatorship createCuratorship(Curatorship curatorship);

    Curatorship updateCuratorship(Curatorship curatorship);

    Boolean deleteCuratorship(Long id);

    void deleteByClientId(Long clientId);

}
